### ISSUE TITLE
What is the focus of Manuskript?

### ISSUE THREAD

**Issue by TheShadowOfHassen** [olivierkes/manuskript#1131]
This is a rather strange question but I'm wondering. What is the focus of Manuskript? What exactly are we aiming at for features? The Readme describes it as "An open source tool for writers." But what kind of writers? I originally assumed for novel writers but there have been feature requests/  features implemented for other sorts of writing.

Take for example the simple mode. According to the blog post, oliverkes meant it so documentation writers could use Manuskript. But with the current feature set I see no difference than just using LibreOffice.

This is kind of important because it basically steers the entire program. If we try to  include all writing (StoryProse, Screenplay, Video Game Essay) Manuskript could easily become a monstrosity that no one could understand and if we try to be too general we'll end up with a text editor and goodness knows there are to many.

Right now it is even more important with the Gtk Port because this is the easiest time to make drastic changes to the layout.

It seems that @TheJackiMonster is the main developer for Manuksript so clearly he'd have a few things to say about this but. I'm curious about what everyone else thinks it is and what they would like to change/ add.

(My thoughts on Manuksript)

I think that primarily Manuksript should be a general fiction writing program for prose.(That's what it feels like and what I thought it was when I first started using it) It should be respectful of most people's different writing styles. (I.E outlining, discovery,) I like to be able to keep track of my characters easily. I don't use the plots very much but I can see how that can be useful.

I created this issue made mostly because of all the feature requests of making the world tab more specific (Adding locations or settings or stuff like that) I think that keeping with the general mindset, the world tab is fine for the way it is.  I don't use it because I write mostly fantasy and I like a bit more features in my world building (I use obsidian) but for someone like a historical fiction writer or a romance writer they don't need that much if any at all.

I don't think that ideas like places and other tabs are a bad idea for people that use them. But I think in a whole Manuksript should be kept a bit more general and let plugins do the rest. (I still want to add plugins eventually!!)

Sorry if this issue is just stating the obvious but I'm seeing conflicted views.

**Comment by TheJackiMonster** [comment]
I personally think Manuskript shouldn't try to concurrent with Office suites. The biggest advantage of Manuskript is that you can plot, write and organize your writing project in one application. So in my opinion it's not about providing the most useful formatting options or the most convenient writing tool but to allow organizing portions of your work with metadata, references, writing targets/goals and a custom tree structure. Besides that the plotting functionality is quite important as well but I think it depends on the individual user how important it is.

Looking at it from a developers point of view, Manuskript reminds me to an IDE for novels/stories instead of software and I think that's a direction we should head for. Making the application more modular sounds good to be honest but we have to keep in mind that the current tabs are strongly connected to the project file structure as well. So making a tab for world objects or characters optional would require changes in the data structure as well to allow more loose references and such.

But as you said it could be beneficial to quite some users because one of Manuskript strengths is (in theory) the simplicity. We don't rely on complex file formats and still offer a lot of features to stitch multiple simple files together to one export ready document. That is one of the reasons I wanted to improve performance as much. Because theoretically Manuskript should have much less overhead than many other writing software solutions. I also like the modular approach of import/export functionality so that most heavy tasks will be processed by Pandoc.

So I think that is what we should focus on: Keeping the heavy processing outside of the application using third-party solutions in a modular way and sticking to organization/writing of users texts and project metadata as main purpose.

**Comment by TheShadowOfHassen** [comment]
Yes, an IDE for writing which is what I thought. I honestly think the way Manuksript is now is perfect and things can be extended by plugins if you want something different. But I'm curious if anyone else has anything they want.

**Comment by Julvenzor** [comment]
Hello. It is a pleasure to be able to participate here. I am a Linux user, biologist and hobbyist writer from Spain with an interest in publishing professionally. To date, I have been one of the translators of Manuskript into Spanish. I have no right to give advice or recommend changes. However, I would like to share my overall impressions of what would improve and what Manuskript should focus on.

I have been trying for several years to be able to use it as a unique and unconditional tool, but I have not succeeded. And this is not your program's fault, because I haven't managed to get used to Scrivener using a virtual machine with Windows either...

Overall, Manuskript does the job, but in my opinion it lacks some essential features if we compare it with proprietary options. Most of my proposals are based on analyzing what I see best done in Scrivener.

1) Text and frame ("Editor" tab): Manuskript should have a simple word processor that allows you to apply styles, select font, size, bold, italic, underline, and paragraph alignment. In the program configuration, you could add a general configuration for the styles used and apply them automatically to our documents or created sections.

Also, and no less important, in Manuskript there is no frame that allows you to see the margins of the text and where you start to write. I consider this a major usability bug.

2) Modularity: Something that is very well done in Scrivener is that the user can manipulate the organization of the novel and the drafts (the menus present in the program) in order to adapt it to their needs. This includes renaming and rearranging folders, changing icons, and adding metadata and notes for different items. This makes it possible, for example, to keep all the books in a series in the same project. Manuskript, however, seems more "hard coded" in the sense that the data and description of the work ("General" and "Summary" tabs) are limited to one volume.

3) Drafts: Another factor that makes Scrivener successful is that he allows himself to be a "game box". The user can insert images, audio files and link notes or semi-written paragraphs to the different written parts in the novel. Manuskript can do some of these things, but it is more limited.

4) Templates and tables: Secondarily, it would be interesting if Manuskript integrated templates to create environments, characters, plots. And, of course, optional templates to submit drafts to publishers with the insertion of values such as author, date, etc.

PS: I just tried the new version and it doesn't stop crashing (segmentation fault) on Kubuntu 22.04. And there are already translated strings in Spanish that continue to be displayed in English. I will open an issue for each bug with more data as soon as I can. I would also like to have some free time to talk about GTK and argue why it would be an absolute bad idea to port the program to this framework, even for Linux users.

In short: GTK only thinks of the Gnome environment, its developers don't listen to the community and seek to create an Apple-style warded garden in which they impose stupid design rules and arbitrary changes that are a pain. Just read the number of developers who are upset by arbitrary changes in GTK 4. There is no future in GTK than in Qt. For me, KDE is the only environment that can succeed on the Linux desktop thanks to powerful third-party technologies.

Regards.

**Comment by TheJackiMonster** [comment]
@Julvenzor First of all thanks for the feedback. Templates are already on their way to be developed. Other features like formatting and adding complex content like images will likely come at some point but it does not have priority at the moment. Also I recommend checking out my explanation why Qt likely gets dropped for development: https://github.com/olivierkes/manuskript/issues/1140#issuecomment-1458980405

I won't support any ideologic discussions here though. GTK gets used on many different desktop environments as well as Qt. Both have different advantages and disadvantages. If people here think this decision has anything to do with some weird desktop environment flame war, I simply can't help you. There are multiple technical reasons for this decision and I also asked for other options here: https://github.com/olivierkes/manuskript/issues/836

Qt is definitely one cause of issues at the moment and the fact that Qt respects a user's theme is just one part of that: https://github.com/olivierkes/manuskript/issues?q=is%3Aopen+is%3Aissue+label%3Aqt-themes

**Comment by Julvenzor** [comment]
Hello. I think that I have been misunderstood. I have only provided my view, not a technical one, that due to the policies I am aware of in these development groups, there may be the possibility of future technical difficulties that make the enormous effort of changing the graphical framework not worth it.

Although ideologies are not reasons to judge the usefulness of a tool (we must separate the author from their work, as I always say), ideologies must be considered to analyze how such tools will evolve and the advantages and disadvantages that they may show in the future due to the "ideals" of those who build and improve it.

The easiest reason I can think of for showing this is that while Qt seeks to be cross-platform for commercial purposes, GTK seeks to serve the needs of a specific niche that doesn't even encompass the Linux community as a whole.

I do not have any hatred towards GTK, GNOME nor do I seek to start an ideological war here. However, while I have been allowed to contribute translations or opinions in many development groups, GTK and Gnome have been the only ones in which I have felt absolute contempt for users and have closed threads with a "won't fix" before changes that have affected me a lot as a writer.

The thread below is not mine, but serves as an example:

"Showing the preedit for compose sequences may be distracting"

https://gitlab.gnome.org/GNOME/gtk/-/issues/3669

Their arguments are like: we do what we want, if you don't like it, leave; plagiarizing Apple is always the best; and the traditional desktop is dead, we are the genius innovators of Linux....

Regards.

**Comment by TheJackiMonster** [comment]
> I do not have any hatred towards GTK, GNOME nor do I seek to start an ideological war here. However, [...]

If you really do not then just stop it. I have nothing else to say but if you don't stop with such non-sense here, I can promise you that I won't continue contributing or maintaining. Because I won't support this ideological madness and I don't have time for this stuff. I can also promise you that KDE developers will dislike what you are doing here as well.

**Comment by TheJackiMonster** [comment]
You don't criticize picking GTK as an option. You imply without any valid sources that this would lead to technical difficulties. You state that GNOME developers would not take issues seriously or not allow contributions, heavily generalizing. We are talking here about free software and voluntary developers have every right to accept or deny any request from users.

To me this has nothing to do with hating GNOME or preferring KDE but disrespecting developers who contribute to free software and if you want to do that. This is not the place for that.

**Comment by Julvenzor** [comment]
I have not disrespected anyone. Criticizing can be fair or unfair, valid or invalid, argued or irrelevant; but it can never imply a lack of respect.

You are the expert, do what you think is appropriate.

**Comment by TheJackiMonster** [comment]
@Julvenzor First of all you ignored my statement to not open ideological discussions here despite this issue being not about it at all, disrespecting me personally and potentially others. Then you continued to allege all GNOME and GTK developers (disrespecting them) to have a negative behavior your own source does not even confirm via deduction which would still be logical non-sense.

So whatever you think you are doing here after your fair points listed in your first comment, it has nothing to do with critic and I don't think such behavior should be supported by any FOSS project out there. Because it doesn't help anyone.

If you have a personal problem with some GNOME developer/contributor, go and solve your issue with them. If you had bad experiences with their projects, I'm sorry. But this has nothing to do with Manuskript.

**Comment by Julvenzor** [comment]
I have no problem with anyone. If you don't want to discuss this, you wouldn't be repeatedly lying and accusing me of disrespecting the developers or having personal issues with someone.

"disrespecting me personally and potentially others".

Are you OK? Do you need to take a lime? I don't think @TheShadowOfHassen was offended by any comments I've made because I haven't even mentioned any of you or judged your role as developers at any point.

I don't care if you answer again. I'm working now and have better things to do than put up with the immature reactions of a post-college kid.
